### PR TITLE
Add `dataset_revision` argument to `RagConfig`

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

The recent changes (yesterday) on the dataset [wiki_dpr](https://huggingface.co/datasets/wiki_dpr/commits/main) cause failing tests.

This PR just uses a revision to avoid the failures. 

Checked on runner and everything is back to the previous status.

### CODE REVIEW

File: src/transformers/models/rag/configuration_rag.py [hunk 1]
```diff
@@ -156,6 +157,7 @@ def __init__(
         self.passages_path = passages_path
         self.index_path = index_path
         self.use_dummy_dataset = use_dummy_dataset
+        self.dataset_revision = dataset_revision
```
Comment (ydshieh) [inline src/transformers/models/rag/configuration_rag.py:160]: well,  I guess it's good to support in general, even if we are not considering testing.

Comment (amyeroberts) [reply src/transformers/models/rag/configuration_rag.py:160]: Agreed. Does this mean that the default configuration for RAG doesn't work, in particular for the examples in the [modeling file](https://github.com/huggingface/transformers/blob/df1542581ee89107eea0569ee044fa8797b66ab0/src/transformers/models/rag/modeling_rag.py#L799), or is it just affecting exact values in the tests?

Comment (ydshieh) [reply src/transformers/models/rag/configuration_rag.py:160]: Good catch! The linked example won't work, but it's not due to a change in `transformers`.

Regarding
> the default configuration doesn't work

I am not really sure if we can talk about `works or not` - when we use `use_dummy_dataset=True`. (well, it's dummy dataset, I don't really know the context of its role in real life usage)

Comment (ydshieh) [reply src/transformers/models/rag/configuration_rag.py:160]: BTW, I pin someone in the dataset team on slack.

Comment (amyeroberts) [reply src/transformers/models/rag/configuration_rag.py:160]: Yep, agree, we still want this feature.

**Review by amyeroberts — APPROVED** [review]
Thanks for adding! 

Could you update the PR title to reflect it's adding a feature rather than fixing?

**Pull request merged**